cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1651,6 +1665,51 @@ String zkConnectionString(int connectToReplicas)  {
                     });
         }
 
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            int replicas = Math.min(oldReplicas, kafkaCluster.getReplicas());
+            List<Future> configFutures = new ArrayList<>(replicas);
+
+            for (int podId = 0; podId < replicas; podId++) {
+                int finalPodId = podId;
+                configFutures.add(podOperations.readiness(namespace, KafkaResources.kafkaPodName(name, podId), 1_000, operationTimeoutMs)
+                    .compose(ignore ->
+                            KafkaBrokerConfigurationHelper.adminClient(secretOperations, adminClientProvider, namespace, name, finalPodId)
+                    .compose(ac -> {
+                        if (ac == null) {
+                            kafkaPodsUpdatedDynamically.put(finalPodId, null);
+                            log.warn("{} Could not create admin client.", reconciliation);
+                            return Future.succeededFuture();
+                        } else {
+                            log.debug("{} Determining kafka pod {} dynamic update ability", reconciliation, finalPodId);
+                            return KafkaBrokerConfigurationHelper.getCurrentConfig(vertx, operationTimeoutMs, finalPodId, ac).compose(res -> {
+                                KafkaBrokerConfigurationDiff configurationDiff = new KafkaBrokerConfigurationDiff(res, this.brokerCm, kafkaCluster.getKafkaVersion(), finalPodId);
+
+                                if (!configurationDiff.isEmpty()) {
+                                    AlterConfigsResult alterConfigResult = ac.incrementalAlterConfigs(configurationDiff.getUpdatedConfig(), new AlterConfigsOptions());
+                                    for (Map.Entry entry : alterConfigResult.values().entrySet()) {
+                                        KafkaFuture kafkaFuture = (KafkaFuture) entry.getValue();
+                                        Util.waitFor(vertx, "KafkaFuture to be completed",  "updated", 1_000, operationTimeoutMs, () -> kafkaFuture.isDone());
```

Comment:
This will block in a vertx worker thread, IIUC. We should factor out a helper method which takes a `KafkaFuture` and returns a vertx `Future`. That should simplify resolving this, I think and will probably be useful elsewhere too.

Reply:
Umpf... I am not experienced with `KafkaFuture`s. Are you willing to help me with this?